org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractXtextTests.java

```diff
@@ -0,0 +1,433 @@
+/*******************************************************************************
+ * Copyright (c) 2008, 2022 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtext.ui.testing;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.util.Iterator;
+import java.util.List;
+
+import org.eclipse.emf.common.notify.Adapter;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.resource.Resource.Diagnostic;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.eclipse.xtext.Constants;
+import org.eclipse.xtext.Grammar;
+import org.eclipse.xtext.IGrammarAccess;
+import org.eclipse.xtext.ISetup;
+import org.eclipse.xtext.conversion.IValueConverterService;
+import org.eclipse.xtext.diagnostics.ExceptionDiagnostic;
+import org.eclipse.xtext.formatting.INodeModelFormatter;
+import org.eclipse.xtext.linking.ILinkingService;
+import org.eclipse.xtext.linking.lazy.LazyLinkingResource;
+import org.eclipse.xtext.nodemodel.ICompositeNode;
+import org.eclipse.xtext.nodemodel.LookAheadInfo;
+import org.eclipse.xtext.nodemodel.impl.InvariantChecker;
+import org.eclipse.xtext.parser.IAstFactory;
+import org.eclipse.xtext.parser.IParseResult;
+import org.eclipse.xtext.parser.IParser;
+import org.eclipse.xtext.resource.IResourceFactory;
+import org.eclipse.xtext.resource.XtextResource;
+import org.eclipse.xtext.resource.XtextResourceSet;
+import org.eclipse.xtext.scoping.IScopeProvider;
+import org.eclipse.xtext.serializer.ISerializer;
+import org.eclipse.xtext.testing.GlobalRegistries;
+import org.eclipse.xtext.testing.GlobalRegistries.GlobalStateMemento;
+import org.eclipse.xtext.testing.serializer.SerializerTestHelper;
+import org.eclipse.xtext.ui.testing.util.ResourceLoadHelper;
+import org.eclipse.xtext.util.CancelIndicator;
+import org.eclipse.xtext.util.LazyStringInputStream;
+import org.eclipse.xtext.util.Pair;
+import org.eclipse.xtext.util.Tuples;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+
+import com.google.common.annotations.Beta;
+import com.google.common.base.Joiner;
+import com.google.common.collect.Lists;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Key;
+import com.google.inject.Module;
+import com.google.inject.Provider;
+import com.google.inject.TypeLiteral;
+import com.google.inject.name.Names;
+
+/**
+ * @author Sven Efftinge - Initial contribution and API
+ * @since 2.30
+ */
+@Beta
+public abstract class AbstractXtextTests extends Assert implements ResourceLoadHelper {
```

Comment:
am not sure if we want this as api. @szarnekow what do you think? copy to the tests instead?

Reply:
can you also check if i copy&pasted the correct variant

Reply:
I don't think we've used this base class in recent test cases. I think it's convoluted and was superseded by the `@InjectWith` / `@RegisterExtension`

Reply:
Yes, I think so. C&P& strip down in the test projects.

Reply:
any hints to strip

Reply:
Subsequent work